src/NHibernate/Engine/Loading/CollectionLoadContext.cs

```diff
@@ -145,7 +145,20 @@ public IPersistentCollection GetLoadingCollection(ICollectionPersister persister
 		/// complete. 
 		/// </summary>
 		/// <param name="persister">The persister for which to complete loading. </param>
+		[Obsolete("Please use EndLoadingCollections(ICollectionPersister, HashSet<string>) instead.")]
```

Comment:
On a method with a generated async counterpart, specifying the exact signature causes the message to be inadequate. (I will a push a commit addressing this point.)